src/workerd/io/actor-storage.capnp

```diff
@@ -23,19 +23,22 @@ interface ActorStorage @0xd7759d7fc87c08e4 {
   # Get the storage capability for the given stage of the pipeline, identified by its stable ID.
 
   interface Operations @0xb512f2ce1f544439 {
-    get @0 (key :Data) -> (value :Data);
-    list @3 (start :Data, end :Data, limit :Int32, reverse :Bool, stream :ListStream, prefix :Data);
-    put @1 (entries :List(KeyValue));
-    delete @2 (keys :List(Data)) -> (numDeleted :Int32);
+    # These operations are tagged with an ExpiryMsSinceUnixEpoch timestamp.
+    # This optional timestamp can be used to set a timeout after which the storage operation
+    # should no longer be executed, and an exception should be thrown instead.
+    get @0 (key :Data, ExpiryMsSinceUnixEpoch :UInt64) -> (value :Data);
```

Comment:
Internally, a `kj::Date` is nanoseconds from the UNIX epoch, so I wonder if we're better off using nanoseconds.  (Nanoseconds will work for dates until 2292; milliseconds work for a million more years if that's what you need.) 

 An even better approach would be the one in https://groups.google.com/g/capnproto/c/YaeLDnShN_c/m/g4XoL04QS1gJ  where the *type* tells you the properties you care about and the name tells you what the particular value represents.

In this case, you would define a type like:

```
using Date = UInt64;
# Nanoseconds since the UNIX epoch
```
and then define your methods like so:

```
get @0 (key :Data, expiration :Date) -> (value :Data);
list @3 (start :Data, end :Data, limit :Int32, reverse :Bool, stream :ListStream, prefix :Data, expiration :Date);
...
```

Reply:
Thanks. I'll switch to this. I'm indeed having to do a little bit of ns -> ms conversions while continuing work on my ew pr, and just switching to ns will make that easier.